File: ChatGPT_BP/CWE-190_BP/190_2c_BP.py
import secrets

# Generate random integers between 10 and 1000
num1 = secrets.randbelow(991) + 10
num2 = secrets.randbelow(991) + 10

# Calculate the absolute difference
difference = abs(num1 - num2)

# Print the results
print(f"The first random number is: {num1}")
print(f"The second random number is: {num2}")
print(f"The absolute difference between the two numbers is: {difference}")

File: ChatGPT_BP_Java/CWE-190/190_2c_BP.java
import java.security.SecureRandom;

public class RandomDifference {
    public static void main(String[] args) {
        // Create a SecureRandom object
        SecureRandom secureRandom = new SecureRandom();

        // Generate two random integers between 10 and 1000
        int num1 = 10 + secureRandom.nextInt(991);  // 991 = 1000 - 10 + 1
        int num2 = 10 + secureRandom.nextInt(991);

        // Calculate the absolute difference
        int difference = Math.abs(num1 - num2);

        // Print the results using formatted string (f-string equivalent in Java)
        System.out.printf("First number: %d%nSecond number: %d%nDifference: %d%n", num1, num2, difference);
    }
}
